### DISCUSSION TITLE
Shadowsocks Client version of a Gluetun container

### DISCUSSION THREAD

**Discussion by glitch452** [qdm12/gluetun#1404] [Q&A] [answered]
I'm wondering if anyone knows of (or maybe this would be a good idea to build) a container that works similar to the Gluetun container, but instead of being a VPN Client, it is a shadowsocks client. It would have to have a similar configuration as Gluetun where it forces all network traffic in the container to go through the shadowsocks connection.  

The general idea being that I could have a single Gluetun container running to connect to my VPN provider, and connect other containers to the Gluetun container via a shadowsocks connection.  

This has two main benefits that I can see:
1. I would only need 1 connection to my VPN provider (since connections are limited, this is beneficial)
2. I could connect containers on other machines on my local network to the Gluetun VPN connection using shadowsocks to connect them

**Comment by securegh** [comment, marked as answer]
> 2\. I could connect containers on other machines on my local network to the Gluetun VPN connection using shadowsocks to connect them

You can use a the [official docker shadowsocks](https://github.com/shadowsocks/shadowsocks-rust) as a client connected to your gluetun shadowsocks server. 
You can connect other docker containers to your shadowsocks docker client like `network_mode: "service:nameofshadowsocksclient"`

hope that helps.

**Comment by ShrirajHegde** [comment]
@securegh Have you tested it? Because there seems to be no firewall and the container does not act as intended.

Shadowsocks client is just supposed to connect to the server and create a local socks5 proxy server. So all network traffic does not go through the proxy server.

@glitch452 were you able to set it up?

**Reply by glitch452** [reply]
I had the same concerns about it not being setup for that so I haven't tried.

**Reply by ShrirajHegde** [reply]
@glitch452 I have made it work using wg-easy container. You can create a Wireguard server that uses VPN. This can then be used with gluetun custom mode with any container as required.

Here is the discussion.
https://github.com/wg-easy/wg-easy/discussions/977#discussioncomment-9132344

Let me know if you want any more information.
